supervisor/api/homeassistant.py

```diff
@@ -66,13 +67,31 @@
 SCHEMA_RESTART = vol.Schema(
     {
         vol.Optional(ATTR_SAFE_MODE, default=False): vol.Boolean(),
+        vol.Optional(ATTR_FORCE, default=False): vol.Boolean(),
+    }
+)
+
+SCHEMA_STOP = vol.Schema(
+    {
+        vol.Optional(ATTR_FORCE, default=False): vol.Boolean(),
     }
 )
 
 
 class APIHomeAssistant(CoreSysAttributes):
     """Handle RESTful API for Home Assistant functions."""
 
+    async def _check_offline_migration(self, force: bool = False) -> None:
+        """Check and raise if there's an offline DB migration in progress."""
```

Comment:
Should we maybe also prevent restart during an online migration? :thinking: 

@emontnemery, opinions?

Reply:
Yes, we should. Core already rejects the service call for shutting down during online migration.

Reply:
Let's do this in a follow up PR. The new feature is offline migration, and this prevents at least restart in that case.